### PR TITLE
feat(graphql) Prevent duplicate glossary term names within a group

### PR DESCRIPTION
**Pull request by chriscollins3456**
Add a check when creating Glossary Terms to ensure that there are no other terms within the same term group (or at the root level if the new term is going to be at the root level) that have the same name.

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-graphql-core/src/test/java/com/linkedin/datahub/graphql/resolvers/glossary/CreateGlossaryTermResolverTest.java [hunk 1]
```diff
@@ -120,4 +137,80 @@ public void testGetSuccessNoParentNode() throws Exception {
...
+    CreateGlossaryTermResolver resolver = new CreateGlossaryTermResolver(mockClient, mockService);
+    assertThrows(CompletionException.class, () -> resolver.get(mockEnv).join());
+
+    Mockito.verify(mockClient, Mockito.times(0)).ingestProposal(
+        Mockito.any(),
+        Mockito.any(Authentication.class)
+    );
+  }
+
+  private EntityClient initMockClient() throws Exception {
```
Comment (joshuaeilers) [inline datahub-graphql-core/src/test/java/com/linkedin/datahub/graphql/resolvers/glossary/CreateGlossaryTermResolverTest.java:192]: You could probably repurpose this for the testGetFailureExistingTermSameName if you support passing in a SearchResult and a result Hashmap

**Review by joshuaeilers — APPROVED** [review]

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/glossary/CreateGlossaryTermResolver.java [hunk 2]
```diff
@@ -100,4 +116,50 @@ private GlossaryTermInfo mapGlossaryTermInfo(final CreateGlossaryEntityInput inp
...
+
+  private Map<Urn, EntityResponse> getTermsWithSameParent(Urn parentNode, QueryContext context) {
+    try {
+      final Filter filter = buildParentNodeFilter(parentNode);
+      final SearchResult searchResult = _entityClient.filter(
+          Constants.GLOSSARY_TERM_ENTITY_NAME,
+          filter,
+          null,
+          0,
+          1000,
```
Comment (joshuaeilers) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/glossary/CreateGlossaryTermResolver.java:134]: Does this mean if there were more than 1000 that we might miss some in this validation?

File: datahub-graphql-core/src/test/java/com/linkedin/datahub/graphql/resolvers/glossary/CreateGlossaryTermResolverTest.java (hunk 1 above)
Comment (chriscollins3456) [reply datahub-graphql-core/src/test/java/com/linkedin/datahub/graphql/resolvers/glossary/CreateGlossaryTermResolverTest.java:192]: oh you right

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/glossary/CreateGlossaryTermResolver.java (hunk 2 above)
Comment (chriscollins3456) [reply datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/glossary/CreateGlossaryTermResolver.java:134]: that is correct. in reality it's extremely unlikely that anyone has more than 1000 entities inside of one term group / folder... but paginating here might make sense just to guard against unexpected use cases.

**Pull request merged**